### ISSUE TITLE
Docker build error

### ISSUE THREAD

**Issue by mhahn** [mozilla-services/heka#1634]
I'm getting the following error when trying to build the docker images:

```
$ ./build_docker.sh
Sending build context to Docker daemon 20.13 MB
Sending build context to Docker daemon
Step 0 : FROM debian:jessie
 ---> 9a61b6b1315e
Step 1 : MAINTAINER Chance Zibolski <chance.zibolski@gmail.com> (@chance)
 ---> Using cache
 ---> e676f9e8ec69
Step 2 : RUN apt-get update &&         apt-get install -yq --no-install-recommends         build-essential         bzr         ca-certificates         cmake         curl         git         golang-goprotobuf-dev        make         mercurial         patch         ruby-dev         protobuf-compiler         python-sphinx         wget
 ---> Using cache
 ---> 9465ef496168
Step 3 : RUN curl -s https://storage.googleapis.com/golang/go1.3.1.linux-amd64.tar.gz -o /tmp/go.tar.gz &&         echo "3af011cc19b21c7180f2604fd85fbc4ddde97143 /tmp/go.tar.gz" | sha1sum -c &&         tar -C /usr/local -xzf /tmp/go.tar.gz
 ---> Using cache
 ---> 268654e46c82
Step 4 : WORKDIR /heka
 ---> Using cache
 ---> 1df8196ef8f5
Step 5 : ENV GOROOT /usr/local/go
 ---> Using cache
 ---> cd6a41c244e9
Step 6 : ENV PATH $PATH:/usr/local/go/bin:/go/bin
 ---> Using cache
 ---> 327396aaa9c3
Step 7 : ENV CTEST_OUTPUT_ON_FAILURE 1
 ---> Using cache
 ---> a8c2fcb5b0ad
Step 8 : ENV BUILD_DIR /heka/build
 ---> Using cache
 ---> ddd4ebbeaa5c
Step 9 : ENV GOPATH $BUILD_DIR/heka
 ---> Using cache
 ---> 4f56c9f152d5
Step 10 : ENV GOBIN $GOPATH/bin
 ---> Using cache
 ---> 7cdf05708f44
Step 11 : ENV PATH $PATH:$GOBIN
 ---> Using cache
 ---> 6e876f6241ae
Step 12 : ENV NUM_JOBS 10
 ---> Using cache
 ---> e9cfb7b8c074
Step 13 : EXPOSE 4352
 ---> Using cache
 ---> 1760aa72225e
Step 14 : COPY . /heka
 ---> Using cache
 ---> 501e0d1d76d4
Step 15 : RUN ./build.sh
 ---> Running in f8879cb87328
-- The C compiler identification is GNU 4.9.2
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
CMake Error at /usr/share/cmake-3.0/Modules/FindPackageHandleStandardArgs.cmake:136 (message):
  Could NOT find Go: Found unsuitable version "1.3.1", but required is at
  least "1.4" (found /usr/local/go/bin/go)
Call Stack (most recent call first):
  /usr/share/cmake-3.0/Modules/FindPackageHandleStandardArgs.cmake:341 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindGo.cmake:32 (find_package_handle_standard_args)
  CMakeLists.txt:18 (find_package)


-- Configuring incomplete, errors occurred!
See also "/heka/build/CMakeFiles/CMakeOutput.log".
make: *** No targets specified and no makefile found.  Stop.
The command '/bin/sh -c ./build.sh' returned a non-zero code: 2
```

**Comment by trink** [comment]
`Could NOT find Go: Found unsuitable version "1.3.1", but required is at   least "1.4" (found /usr/local/go/bin/go)`

**Comment by mhahn** [comment]
I understood what the error was, but it seems like go 1.3.1 is being installed here:

https://github.com/mozilla-services/heka/blob/dev/Dockerfile#L24

**Comment by trink** [comment]
@mhahn Sorry, I thought that was your build env and not something we setup.

**Comment by rafrombrc** [comment]
This should be resolved on the versions/0.10 branch as of #1696 being merged back. Closing this issue, please test and re-open if the problem isn't resolved. Thanks!
